### PR TITLE
[Outlook] - NAA sample with IE fallback

### PR DESCRIPTION
**Pull request by codexeon**
|        Q        |                    A                    |
| --------------- | --------------------------------------- |
| Bug fix?        | no                              |
| New feature?    | no                              |
| New sample?     | yes                              |
| Related issues? | NA |

## What's in this Pull Request?

This adds a new sample that is similar to Outlook-Add-in-SSO-NAA, but has support for Internet Explorer (IE) fallback. IE fallback sample may not be for every app developer as it adds overhead of supporting multiple auth libraries and polyfills, but can be used as a reference if your add-in must support IE environments.

### CODE REVIEW

File: Samples/auth/Outlook-Add-in-SSO-NAA-IE/src/taskpane/fallback/dialog.html [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+<!doctype html>
+<html>
+  <head>
+    <meta charset="UTF-8" />
+    <meta http-equiv="X-UA-Compatible" content="IE=Edge" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <title>Dialog</title>
+
+    <!-- Office JavaScript API -->
+    <script type="text/javascript" src="https://appsforoffice.microsoft.com/lib/1.1/hosted/office.debug.js"></script>
```
Comment (AlexJerabek) [inline Samples/auth/Outlook-Add-in-SSO-NAA-IE/src/taskpane/fallback/dialog.html:13]: Should this not be the debug version?
```suggestion
    <script type="text/javascript" src="https://appsforoffice.microsoft.com/lib/1.1/hosted/office.js"></script>
```

**Review by AlexJerabek — CHANGES_REQUESTED** [review]
The sample looks good. This is mostly a copy edit on the readme. Has someone else tested the IE11 fallback path?

File: Samples/auth/Outlook-Add-in-SSO-NAA-IE/README.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+- You need to join the [Microsoft 365 Insider Program](https://insider.microsoft365.com/join) to use the NAA preview features. Choose the **Current Channel (Preview)** insider level.
+- [Node.js](https://nodejs.org/) version 16 or greater.
+- [npm](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm) version 8 or greater.
+
+## Build and run the solution
+
+### Create an application registration
+
+1. Go to the [Azure portal - App registrations](https://go.microsoft.com/fwlink/?linkid=2083908) page to register your app.
+1. Sign in with the **_admin_** credentials to your Microsoft 365 tenancy. For example, **MyName@contoso.onmicrosoft.com**.
```
[the remark this answers is not in the document]
Comment (codexeon) [reply Samples/auth/Outlook-Add-in-SSO-NAA-IE/README.md:47]: not sure I understand the comment, but this matches existing Outlook-Add-in-SSO-NAA sample. It doesn't require admin to create the registration, but it does require it to consent to files.read scope.

File: Samples/auth/Outlook-Add-in-SSO-NAA-IE/README.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+1. Select **New registration**. On the **Register an application** page, set the values as follows.
+
+   - Set **Name** to `Outlook-Add-in-SSO-NAA`.
```
[the remark this answers is not in the document]
Comment (codexeon) [reply Samples/auth/Outlook-Add-in-SSO-NAA-IE/README.md:50]: it shouldn't be a problem as it has a unique id, but I updated the name.

File: Samples/auth/Outlook-Add-in-SSO-NAA-IE/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (AlexJerabek) [reply Samples/auth/Outlook-Add-in-SSO-NAA-IE/README.md:47]: I read this sentence as "Sign in with your admin account (e.g.: MyName@contoso.onmicrosoft.com)." I'm not sure why the email address help clarify anything. Guess this is more of a general question for @davidchesnut across all the samples. Please leave it as is.

File: Samples/auth/Outlook-Add-in-SSO-NAA-IE/assets/sample.json [hunk 4]
```diff
@@ -1,41 +1,37 @@
...
-          "name": "Enable SSO in an Office Add-in using nested app authentication (preview)",
-          "description": "Learn how to enable SSO in an Office Add-in using nested app authentication.",
-          "url": "https://learn.microsoft.com/office/dev/add-ins/develop/enable-nested-app-authentication-in-your-add-in"
-        }
-      ]
-    }
-  ] 
\ No newline at end of file
+  {
+    "name": "outlook-add-in-sso-naa",
```
Comment (AlexJerabek) [inline Samples/auth/Outlook-Add-in-SSO-NAA-IE/assets/sample.json:3]:
```suggestion
    "name": "outlook-add-in-sso-naa-ie",
```

**Review by AlexJerabek — APPROVED** [review]
Thanks for making those changes. Approving with one more suggestion.

**Pull request merged**